app/views/user/profile.php

```diff
@@ -1,21 +1,92 @@
 <h2>My Profile</h2>
 
-<form class="span6 well" method="POST">
+<?php if ($user->hasError()): ?>
+    <div class="alert alert-block">
+        <h4 class="alert-heading">Registration failed!</h4>
+
+        <?php if ($user->validation_errors['firstname']['length']): ?>
+            <div>
+                <em>First name</em> must be between
+                <?php enquote_string($user->validation['firstname']['length'][1]) ?> and
+                <?php enquote_string($user->validation['firstname']['length'][2]) ?> characters in length.
+            </div>
+        <?php endif ?>
+
+        <?php if ($user->validation_errors['firstname']['confirmation']): ?>
+            <div>
+                <em>First name</em> should contain letters only.
+            </div>
+        <?php endif ?>
+        
+        <?php if ($user->validation_errors['lastname']['length']): ?>
+            <div>
+                <em>Last name</em> must be between
+                <?php enquote_string($user->validation['lastname']['length'][1]) ?> and
+                <?php enquote_string($user->validation['lastname']['length'][2]) ?> characters in length.
+            </div>
+        <?php endif ?>
+
+        <?php if ($user->validation_errors['lastname']['confirmation']): ?>
+            <div>
+                <em>Last name</em> should contain letters only.
+            </div>
+        <?php endif ?>
+    </div>
+<?php endif ?>
+<form class="span6 well" method="POST" style="box-shadow: 10px 10px 10px #888888">
     <label>First name</label>
     <input class="span6" type="text" value="<?php echo $firstname ?>" name="firstname" required>
     <label>Last name</label>
     <input class="span6" type="text" value="<?php echo $lastname ?>" name="lastname" required>
     <label>Username</label>
-    <input class="span6" type="text" value="<?php echo $username ?>" name="username" required>
+    <input class="span6" type="text" value="<?php echo $username ?>" name="username" disabled   >
     <label>Email</label>
-    <input class="span6" type="email" value="<?php echo $email ?>" name="email" required>
+    <input class="span6" type="email" value="<?php echo $email ?>" name="email" disabled>
     <label>Member since</label>
     <input class="span6" type="text" value="<?php echo $member_since ?> ago" name="member_since" disabled>
     <br/>
      <input type="hidden" name="page_next" value="success_update">
     <button type="submit" class="btn btn-danger">Save</button>
+    <a href="<?php enquote_string(url('user/change_password'))?>" class="btn btn-danger">Change Password</a>
 </form>
 
-<form class="offset1 span3 well">
-    Insert avatar here
+<form class="offset1 span3 well" style="box-shadow: 10px 10px 10px #888888">
```

Comment:
this is only a suggestion, you could've create your own css file then put all your inline css in that, in that way you could reuse the styles and make the code neat.